src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+                {
+                    cur_horizons_list.push_back( unit.second.interface_top_ );
+                }
+                if( unit.second.interface_base_.is_defined() )
+                {
+                    cur_horizons_list.push_back( unit.second.interface_base_ );
+                }
+                std::sort( cur_horizons_list.begin(), cur_horizons_list.end() );
+
+                if( cur_horizons_list.size() != horizons_list.size() )
+                {
+                    continue;
+                }
+                std::vector< gmge_id > result;
+                std::set_intersection( horizons_list.begin(),
+                    horizons_list.end(), cur_horizons_list.begin(),
+                    cur_horizons_list.end(), std::back_inserter( result ) );
+                if( result.size() == horizons_list.size() )
+                {
+                    auto id = interp_2_geo_entity_.find( unit.first );
+                    ringmesh_assert( id != interp_2_geo_entity_.end() );
+                    return id->second;
+                }
+            }
+        }
+        return gmge_id();
+    }
+
+    void GeoModelBuilderRESQMLImpl::read_strati_column( EpcDocument& pck )
+    {
+        const std::vector< RESQML2_0_1_NS::StratigraphicColumn* >
+            stratiColumnSet = pck.getStratigraphicColumnSet();
+        // TODO only one column for the moment
+        ringmesh_assert( stratiColumnSet.size() <= 1 );
+
+        for( auto column : stratiColumnSet )
+        {
+            showAllMetadata( column );
+
+            // TODO now we read only rank 0
+            ringmesh_assert(
+                column->getStratigraphicColumnRankInterpretationSet().size()
+                == 1 );
+
+            for( auto rank_interp :
+                column->getStratigraphicColumnRankInterpretationSet() )
+            {
+                showAllMetadata( rank_interp );
+                // TODO only chronostrati
+                ringmesh_assert( rank_interp->isAChronoStratiRank() );
+
+                for( auto unit :
+                    rank_interp->getStratigraphicUnitInterpretationSet() )
+                {
+                    const gmge_id layer =
+                        builder_.geology.create_geological_entity(
+                            Layer3D::type_name_static() );
+
+                    builder_.geology.set_geological_entity_geol_feature(
+                        layer, GMGE::GEOL_FEATURE::STRATI_UNIT );
+
+                    interp_2_geo_entity_[unit] = layer;
+
+                    unit_2_info_[unit].layer_ = layer;
+                    if( unit->getInterpretedFeature() != nullptr )
+                    {
+                        const Layer3D& layer3d = static_cast< const Layer3D& >(
+                            geomodel_.geological_entity(
+                                GeologicalEntityType( "Layer" ),
+                                layer.index() ) );
+                        const_cast< Layer3D& >( layer3d ).set_name(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor the code to not need "const_cast". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS859)